api/src/services/telemetry-services/telemetry-vectronic-service.ts

```diff
@@ -0,0 +1,193 @@
+import axios, { AxiosInstance } from 'axios';
+import { chunk } from 'lodash';
+import qs from 'qs';
+import { TelemetryCredentialVectronicRecord } from '../../database-models/telemetry_credential_vectronic';
+import { IDBConnection } from '../../database/db';
+import { ApiGeneralError } from '../../errors/api-error';
+import { formatAxiosError } from '../../errors/axios-error';
+import { TelemetryVectronicRepository } from '../../repositories/telemetry-repositories/telemetry-vectronic-repository';
+import {
+  VectronicAPIQuery,
+  VectronicPayload,
+  VectronicTask
+} from '../../repositories/telemetry-repositories/telemetry-vectronic-repository.interface';
+import { getEnvironmentVariable } from '../../utils/env-config';
+import { getLogger } from '../../utils/logger';
+import { QueueResult, taskQueue } from '../../utils/task-queue';
+import { DBService } from '../db-service';
+import { keysToLowerCase } from './telemetry-utils';
+import { TelemetryProcessingOptions, TelemetryProcessingResult } from './telemetry.interface';
+const defaultLog = getLogger('telemetry-vectronic-service');
+
+/**
+ * This service is responsible for fetching telemetry data from the Vectronic API and storing it in SIMS.
+ *
+ * @see https://api.vectronic-wildlife.com/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config
+ *
+ * @export
+ * @class TelemetryVendorService
+ * @extends {DBService}
+ */
+export class TelemetryVectronicService extends DBService {
+  vectronicClient: AxiosInstance;
+
+  telemetryVectronicRepository: TelemetryVectronicRepository;
+
+  /**
+   * Creates an instance of TelemetryVectronicService.
+   *
+   * @param {IDBConnection} connection
+   */
+  constructor(connection: IDBConnection) {
+    super(connection);
+
+    this.vectronicClient = axios.create({
+      paramsSerializer: (params) => qs.stringify(params),
+      baseURL: `${getEnvironmentVariable('VECTRONIC_API_HOST')}/v2`
```

Comment:
Should the `/v2` portion just be included the env var?

Reply:
Updated